src/app/pages/blog/search-results/use-all-articles.js

```diff
@@ -0,0 +1,28 @@
+import {useState, useEffect} from 'react';
+import {useLocation} from 'react-router-dom';
+import {fetchFromCMS, camelCaseKeys} from '~/helpers/page-data-utils';
+import {blurbModel} from '../article-summary/article-summary';
+import uniqBy from 'lodash/uniqBy';
+
+export default function useAllArticles() {
+    const {search} = useLocation();
+    const searchParam = new window.URLSearchParams(search).get('q');
+    const [allArticles, setAllArticles] = useState([]);
+
+    useEffect(() => {
+        const slug = `search/?q=${searchParam}`;
+
+        setAllArticles([]);
+        fetchFromCMS(slug, true).then((results) => {
+            setAllArticles(
+                uniqBy(results, 'id').map((data) => {
+                    data.heading = data.title;
+                    delete data.subheading;
+                    return blurbModel(camelCaseKeys(data));
+                })
+            );
+        });
+    }, [searchParam]);
```

Comment:
This is from the old code, so maybe all is good, but at a glance I would have thought `search` would be need to be here instead?

Reply:
`searchParam` is what's used in the effect. It is dependent on `search` and not used outside it, so I could create it in the effect and make that the dependency.